crates/code_generation/src/cyclers.rs

```diff
@@ -415,6 +420,9 @@ fn generate_cycle_method(cycler: &Cycler, cyclers: &Cyclers) -> TokenStream {
                     own_database.deref_mut()
                 };
 
+                let enable_recording = <HardwareInterface as hardware::RecordingInterface>::get_recording(&*self.hardware_interface);
+                let mut recording_frame = Vec::new(); // TODO: possible optimization: cache capacity
```

Comment:
It seems, I forgot to serialize the nodes. The idea was to do this at the beginning. This would be the information needed for replay. Currently, I would try a simple implementation by serializing every frame. Let's see, if we need to optimize.